### PR TITLE
 Water Sensor shown on display with icons

### PR DESCRIPTION
**Pull request by genkigenki**
Water Sensor, "water empty" icon shown in top right, on each display mode
In case the water sensor is not configured, water_full=true always, so no icon will ever be shown.

### CODE REVIEW

**Review by fiendie — CHANGES_REQUESTED** [review]

File: src/display.h [hunk 1]
```diff
@@ -213,12 +213,22 @@ void Displaymachinestate() {
         u8g2.setFont(u8g2_font_profont17_tf);
         u8g2.print(temperature, 1);
         u8g2.sendBuffer();
+
+        //Small water empty logo
+        if (waterFull == false) {
+            //small water empty logo
+            u8g2.drawXBMP(2, 2, 8, 8, water_EMPTY_u8g2);
+		}
```
[the remark this answers is not in the document]
Comment (genkigenki) [reply src/display.h:221]: looks good with VS code
grafik

File: src/main.cpp [hunk 2]
```diff
@@ -2383,6 +2405,43 @@ void loopLED() {
...
+            if (isWaterDetected) {
+                // Water is detected, increment counter if it was previously empty
+                if (!waterFull) {
+                    WaterCheckConsecutiveReads++;
+                    if (WaterCheckConsecutiveReads >= WaterCountsNeeded) {
+                        waterFull = true;
+                        debugPrintf("Water full\n");
+                        WaterCheckConsecutiveReads = 0; // Reset counter
+                    }
+                } else {
```
[the remark this answers is not in the document]
Comment (genkigenki) [reply src/main.cpp:2424]: ok done at all 3 positions

File: src/main.cpp [hunk 3]
```diff
@@ -2383,6 +2405,43 @@ void loopLED() {
...
+                    WaterCheckConsecutiveReads = 0; // Reset counter if water was already full
+                }
+            } else {
```
[the remark this answers is not in the document]
Comment (genkigenki) [reply src/main.cpp:2427]: Resolved.
However, in the whole code there are many instances like this.

File: src/display.h [hunk 4]
```diff
@@ -229,6 +239,10 @@ void Displaymachinestate() {
         u8g2.clearBuffer();
         u8g2.drawXBMP(38, 0, OFFLogo_width, OFFLogo_height, OFFLogo);
         u8g2.setCursor(36, 55);
+        if (waterFull == false) {
```
[the remark this answers is not in the document]
Comment (genkigenki) [reply src/display.h:243]: done

File: src/main.cpp [hunk 5]
```diff
@@ -2040,6 +2049,16 @@ void setup() {
         pinMode(PIN_STATUSLED, OUTPUT);
     }
 
+    #if WATER_SENS == 1
+        //NPN
+        pinMode(PIN_WATERSENSOR, INPUT_PULLUP);
+    #endif
+
```
Comment (fiendie) [inline src/main.cpp:2056]: Can be shortened to #if ... #elif ... #endif
I know it's not consistent in the codebase but while we're at it :)

**Review by fiendie — CHANGES_REQUESTED** [review]

**Review by fiendie — APPROVED** [review]

**Review by fiendie — APPROVED** [review]

**Pull request merged**